src/chainrules.jl

```diff
@@ -19,9 +19,15 @@ end
 function _Ω_∂_xlogy(x::Real, y::Real)
     logy = log(y)
     z = x * logy
-    Ω = iszero(x) && !isnan(y) ? zero(z) : z
-    ∂x = logy
-    ∂y = x / y
+    w = x / y
+    if iszero(x) && !isnan(y)
+        Ω = zero(z)
+        ∂y = zero(w)
```

Comment:
This is what I had in mind and think should be added. I hadn't managed to open a PR since I'm on vacation until beginning of August and only have limited time right now 🙂

Reply:
Great!